sentry/src/main/java/io/sentry/TraceContext.java

```diff
@@ -50,8 +59,39 @@ public final class TraceContext implements JsonUnknown, JsonSerializable {
         new Dsn(sentryOptions.getDsn()).getPublicKey(),
         sentryOptions.getRelease(),
         sentryOptions.getEnvironment(),
-        user != null ? new TraceContextUser(user) : null,
-        transaction.getName());
+        user != null ? user.getId() : null,
+        user != null ? getSegment(user) : null,
+        transaction.getName(),
+        sampleRateToString(sentryOptions.getTracesSampleRate()));
+  }
+
+  private static @Nullable String getSegment(final @NotNull User user) {
+    final Map<String, String> others = user.getOthers();
+    if (others != null) {
+      return others.get("segment");
+    } else {
+      return null;
+    }
+  }
+
+  private static @Nullable String sampleRateToString(@Nullable Double sampleRateAsDouble) {
+    if (sampleRateAsDouble == null) {
+      return null;
+    }
+
+    if (sampleRateAsDouble == null
+        || sampleRateAsDouble.isNaN()
+        || sampleRateAsDouble.isInfinite()) {
+      return null;
+    }
+
+    if (sampleRateAsDouble < 0.0 || sampleRateAsDouble > 1.0) {
+      return null;
+    }
```

Comment:
Maybe we extract this logic from `SentryOptions#setSampleRate` and `SentryOptions#setTracesSampleRate` and reuse it as an util function.

Reply:
`setSampleRate` has `sampleRate > 1.0 || sampleRate <= 0.0` while `setTracesSampleRate` has `tracesSampleRate > 1.0 || tracesSampleRate < 0.0`. Can add a util but won't reuse in both.

Reply:
Added `SampleRateUtil` in #2141